src/content/blog/2024-12-04-differences-between-linting-and-type-checking.md

```diff
@@ -0,0 +1,248 @@
+---
+layout: post
+title: "Differences between linting and type checking"
+teaser: "Linters such as ESLint and type checkers such as TypeScript catch different areas of code defects — and are best used in conjunction with each other."
+authors:
+  - joshuakgoldberg
+categories:
+  - Storytime
+tags:
+  - Linting
+---
+
+Linters and type checkers are two kinds of tools that both analyze code and report on detected issues.
+They may seem similar at first.
+However, the two kinds of tools detect very different issues and are useful in different ways.
+
+## Definitions
+
+*"Static analysis"* is the term for tooling that checks code for issues without running it.
+There are three forms of static analysis commonly used in web development projects today:
+
+1. **Formatters**: only print your code quickly, without worrying about logic
+2. **Linters**: execute individually configurable checks known as "lint rules"
+3. **Type checkers**: collect all files into a full understanding of the project
+
+We'll focus in this blog post on *linters*, namely ESLint, and *type checkers*, namely [TypeScript](https://typescriptlang.org).
+ESLint and TypeScript are often used together to detect defects in code.
+
+## Differences in Purpose
+
+Type checkers make sure the *intent* behind values in code matches the *usage* of those values.
+They check that code is "type-safe": meaning values are used according to the ways their types describe as being allowed.
+
+Type checkers do not attempt to look for defects that are only likely, rather than certain.
```

Comment:
This sentence is difficult to parse. I'm not sure what it's supposed to be conveying.